airflow/www/static/js/api/useSetDagRunNotes.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import axios from 'axios';
+import { useMutation, useQueryClient } from 'react-query';
+
+import { getMetaValue } from 'src/utils';
+import type { API } from 'src/types';
+import useErrorToast from 'src/utils/useErrorToast';
+
+import { emptyGridData } from './useGridData';
+import type { GridData } from './useGridData';
+
+const setDagRunNotesURI = getMetaValue('set_dag_run_notes');
+
+interface Props {
+  dagId: string;
+  runId: string;
+}
+
+export default function useSetDagRunNotes({
+  dagId, runId,
+}: Props) {
+  const queryClient = useQueryClient();
+  const errorToast = useErrorToast();
+  const setDagRunNotes = setDagRunNotesURI.replace('_DAG_RUN_ID_', runId);
+
+  return useMutation(
+    ['setDagRunNotes', dagId, runId],
+    (notes: string | null) => axios.patch(setDagRunNotes, { notes }),
+    {
+      onSuccess: async (data) => {
+        const notes = (data as API.DAGRun).notes ?? null;
```

Comment:
So you can avoid casting here
```suggestion
        const notes = data.notes ?? null;
```